tests/v2/test_waku_noise.nim

```diff
@@ -2,22 +2,115 @@
 
 import
   testutils/unittests,
+  std/random,
+  stew/byteutils,
   ../../waku/v2/protocol/waku_noise/noise,
   ../test_helpers
 
-
 procSuite "Waku Noise":
   
   let rng = rng()
 
-  test "ChaChaPoly Encryption/Decryption":
+  test "ChaChaPoly Encryption/Decryption: random byte sequences":
 
     let cipherState = randomChaChaPolyCipherState(rng[])
 
+    # We encrypt/decrypt random byte sequences
     let
-      plaintext: seq[byte] = randomSeqByte(rng[], 128)
+      plaintext: seq[byte] = randomSeqByte(rng[], 64)
       ciphertext: ChaChaPolyCiphertext = encrypt(cipherState, plaintext)
       decryptedCiphertext: seq[byte] = decrypt(cipherState, ciphertext)
 
     check: 
-      plaintext == decryptedCiphertext
\ No newline at end of file
+      plaintext == decryptedCiphertext
+
+  test "ChaChaPoly Encryption/Decryption: random strings":
+
+    let cipherState = randomChaChaPolyCipherState(rng[])
+
+    # We encrypt/decrypt random strings
+    randomize()
+    var plaintext: string
+    for _ in .. rand(64):
+      add(plaintext, char(rand(int('A') .. int('z'))))
+
+    echo plaintext
+
+    let
+      ciphertext: ChaChaPolyCiphertext = encrypt(cipherState, plaintext.toBytes())
+      decryptedCiphertext: seq[byte] = decrypt(cipherState, ciphertext)
+
+    check: 
+      plaintext.toBytes() == decryptedCiphertext
+
+  test "Encrypt -> decrypt Noise public keys":
+
+    let noisePublicKey: NoisePublicKey = genNoisePublicKey(rng[])
+
+    let 
+      cs: ChaChaPolyCipherState = randomChaChaPolyCipherState(rng[])
+      encryptedPk: NoisePublicKey = encryptNoisePublicKey(cs, noisePublicKey)
+      decryptedPk: NoisePublicKey = decryptNoisePublicKey(cs, encryptedPk)
+
+    check: 
+      noisePublicKey == decryptedPk
+
+  test "Decrypt unencrypted public key":
+
+    let noisePublicKey: NoisePublicKey = genNoisePublicKey(rng[])
+
+    let 
+      cs: ChaChaPolyCipherState = randomChaChaPolyCipherState(rng[])
+      decryptedPk: NoisePublicKey = decryptNoisePublicKey(cs, noisePublicKey)
+
+    check:
+      noisePublicKey == decryptedPk
+
+  test "Encrypt -> encrypt public keys":
```

Comment:
Do you mean, encrypt encrypted public keys?

Reply:
Yes, it tests double encryption. The second call should return the public key as it is without encrypting once more time. I'll make it clearer